dtde/config.py

```diff
@@ -26,3 +27,45 @@ def get() -> configparser.ConfigParser:
     if not global_config:
         load_config()
     return global_config
+
+
+class ConfigObject:
+
+    def __init__(self):
+        self.cluster_id = None
+        self.cluster_vm_size = None
+        self.cluster_size = None
+        self.cluster_size_low_pri = None
+        self.cluster_username = None
+        self.cluster_password = None
+        self.cluster_ssh_key = None
+        self.custom_script = None
+        self.wait = None
+
+
+    def read_config_file(self, path: str=constants.DEFAULT_CLUSTER_CONFIG_PATH):
+        """
+            Reads the config file in the config/ directory (config/cluster.yaml)
+        """
+        if not os.path.isfile(path):
+            raise Exception(
+                    "Configuration file doesn't exist at {0}".format(path))
+
+        with open(path, 'r') as stream:
+            try:
+                config = yaml.load(stream)
+                self.cluster_id = config['base']['cluster_id']
+                self.cluster_vm_size = config['base']['cluster_vm_size']
+                self.cluster_size = config['base']['cluster_size']
+                if self.cluster_size is None:
+                    self.cluster_size = 0
+                self.cluster_size_low_pri = config['base']['cluster_size_low_pri']
+                if self.cluster_size_low_pri is None:
+                    self.cluster_size_low_pri = 0
+                self.cluster_username = config['base']['cluster_username']
+                self.cluster_password = config['base']['cluster_password']
+                self.cluster_ssh_key = config['base']['cluster_ssh_key']
+                self.custom_script = config['base']['custom_script']
+                self.wait = config['base']['wait']
+            except ymal.YAMLError as err:
```

Comment:
we should probably do more error handling here right?

if you specify both size-low-pri and size, we should stop the process and spit out a msg saying that it failed

Reply:
+1. same goes for password and ssh_key. Oh, and also if they want a username, should we warn them if WAIT == false? (that last one should be a no-op once the service supports this feature for us)